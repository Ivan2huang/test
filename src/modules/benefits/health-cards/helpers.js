/* eslint-disable import/prefer-default-export */
export const getQRCodeXML = ({
  fullName,
  membershipNumber,
  certificateNumber,
  insurerCode,
  insurerName,
  insurerType,
  policyNumber,
  expiryDate,
}) =>
  [
    '<?xml version="1.0"?>',
    '<medical_card>',
    '<Card_type>',
    insurerType,
    '</Card_type>',
    '<insurance_company_code>',
    insurerCode,
    '</insurance_company_code>',
    '<insurance_company_name>',
    insurerName,
    '</insurance_company_name>',
    '<insured_name>',
    fullName,
    '</insured_name>',
    '<policy_no>',
    policyNumber,
    '</policy_no>',
    '<certificate_no>',
    certificateNumber,
    '</certificate_no>',
    '<membership_no>',
    membershipNumber,
    '</membership_no>',
    '<expiry_date>',
    expiryDate,
    '</expiry_date>',
    '</medical_card>',
  ].join('');
