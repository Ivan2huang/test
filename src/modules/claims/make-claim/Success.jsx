import React from 'react';
import * as PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import { compose } from 'redux';

import { Box, Card, CardContent, withStyles, Button } from '@material-ui/core';

import Images from '../../../constants/images';
import { formatMessage, navigateTo } from '../../../helpers/helpers';
import paths from '../../../helpers/paths';
import Typography from '../../../uiComponents/Typography';
import ButtonGroup from '../../../uiComponents/ButtonGroup';
import Grid from '../../../uiComponents/Grid';
import GridItem from '../../../uiComponents/GridItem';

const Styles = () => ({
  image: {
    width: '100%',
  },
});

const Success = ({ intl, classes }) => {
  const navigateToClaims = () => {
    navigateTo(paths.common.claims);
  };
  const navigateToMakeClaim = () => {
    navigateTo(paths.employee.makeClaim);
  };

  return (
    <Box
      mt={{
        xs: 3,
        sm: 14,
      }}
    >
      <Card>
        <CardContent>
          <Box mt={10} mb={10}>
            <Grid>
              <GridItem offset={{ md: 1 }} columns={{ md: 3 }}>
                <img
                  src={Images.SUCCESS_CLAIM_BACKGROUND}
                  alt="success claim background"
                  className={classes.image}
                />
              </GridItem>
              <GridItem
                offset={{ sm: 1, md: 1 }}
                columns={{ xs: 12, sm: 10, md: 5 }}
              >
                <Box mb={8}>
                  <Typography type="style-1">
                    {formatMessage(
                      intl,
                      'claims.makeClaimSuccess.title',
                      'Claim submitted',
                    )}
                  </Typography>
                </Box>
                <Box mb={8}>
                  <Typography type="style-6">
                    {formatMessage(
                      intl,
                      'claims.makeClaimSuccess.description',
                      'Thank you. Please keep your original documents for at least 6 months. Your claim will be settled in 14 days.',
                    )}
                  </Typography>
                </Box>
                <Box mb={8}>
                  <Typography type="style-6">
                    {formatMessage(
                      intl,
                      'claims.makeClaimSuccess.disclaimer',
                      'Your claim might take 2-3 days before it shows up in the claims list.',
                    )}
                  </Typography>
                </Box>
                <ButtonGroup>
                  <Button
                    data-testid="btn-back-to-claims"
                    variant="contained"
                    color="primary"
                    onClick={navigateToClaims}
                  >
                    {formatMessage(
                      intl,
                      'claims.makeClaimSuccess.button.backToClaims',
                      'Back to claims',
                    )}
                  </Button>
                  <Button
                    data-testid="btn-make-another-claim"
                    variant="outlined"
                    color="secondary"
                    onClick={navigateToMakeClaim}
                  >
                    {formatMessage(
                      intl,
                      'claims.makeClaimSuccess.button.makeAnotherClaim',
                      'Make another claim',
                    )}
                  </Button>
                </ButtonGroup>
              </GridItem>
            </Grid>
          </Box>
        </CardContent>
      </Card>
    </Box>
  );
};

Success.propTypes = {
  intl: PropTypes.shape({}).isRequired,
  classes: PropTypes.exact({
    image: PropTypes.string.isRequired,
  }).isRequired,
};

export default compose(
  injectIntl,
  withStyles(Styles),
)(Success);
