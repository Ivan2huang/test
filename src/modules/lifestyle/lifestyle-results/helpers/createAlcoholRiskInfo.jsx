import React from 'react';

import { AlcoholIcon, NoAlcoholIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createAlcoholRiskInfo = (intl, theme, alcoholRisk) => {
  const alcoholRiskMapping = {
    None: `${theme.lifeStyleResultOK}`,
    RegularDrinker: `${theme.lifeStyleResultWarning}`,
    HeavyDrinker: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'alcohol',
    color: alcoholRiskMapping[alcoholRisk],
    CardIcon:
      alcoholRisk === 'None' ? (
        <NoAlcoholIcon />
      ) : (
        <AlcoholIcon color={alcoholRiskMapping[alcoholRisk]} />
      ),
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.alcoholRiskCard.title',
      'Alcohol',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.alcoholRiskCard.subTitle.${alcoholRisk}`,
      alcoholRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.alcoholRiskCard.description.${alcoholRisk}`,
      alcoholRisk,
    ),
  };
};

export default createAlcoholRiskInfo;
