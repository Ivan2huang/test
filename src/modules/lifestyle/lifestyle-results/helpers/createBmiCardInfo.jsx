import React from 'react';

import Typography from '../../../../uiComponents/Typography';
import { formatMessage } from '../../../../helpers/helpers';

export const formatBmiScore = bmiScore => {
  const score = parseFloat(bmiScore);
  if (Number.isNaN(score)) {
    return '';
  }
  return (+bmiScore).toFixed(1);
};

const createBmiCardInfo = (intl, theme, bmiScore, bmiClass) => {
  const bmiColorMapping = {
    Healthy: `${theme.lifeStyleResultOK}`,
    Overweight: `${theme.lifeStyleResultWarning}`,
    Obese: `${theme.lifeStyleResultAlarm}`,
    Underweight: `${theme.lifeStyleResultWarning}`,
  };

  return {
    id: 'bmi',
    color: bmiColorMapping[bmiClass],
    CardIcon: (
      <Typography type="style-1" color={theme.grey3} fontWeight="bold">
        {formatBmiScore(bmiScore)}
      </Typography>
    ),
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.bmiCard.title',
      'BMI',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.bmiCard.subTitle.${bmiClass}`,
      bmiClass,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.bmiCard.description.${bmiClass}`,
      bmiClass,
    ),
  };
};

export default createBmiCardInfo;
