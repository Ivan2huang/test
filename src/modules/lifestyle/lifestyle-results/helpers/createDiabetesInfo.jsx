import React from 'react';

import { HeartIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createDiabetesInfo = (intl, theme, diabetesRisk) => {
  const alcoholRiskMapping = {
    Low: `${theme.lifeStyleResultOK}`,
    High: `${theme.lifeStyleResultWarning}`,
    VeryHigh: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'diabetes',
    color: alcoholRiskMapping[diabetesRisk],
    CardIcon: <HeartIcon color={alcoholRiskMapping[diabetesRisk]} />,
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.diabetesCard.title',
      'Diabetes',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.diabetesCard.subTitle.${diabetesRisk}`,
      diabetesRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.diabetesCard.description.${diabetesRisk}`,
      diabetesRisk,
    ),
  };
};

export default createDiabetesInfo;
