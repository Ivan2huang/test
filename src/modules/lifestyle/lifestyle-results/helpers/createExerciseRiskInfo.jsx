import React from 'react';

import { ExerciseIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createExerciseRiskInfo = (intl, theme, exerciseRisk) => {
  const exerciseRiskMapping = {
    High: `${theme.lifeStyleResultOK}`,
    LowToModerate: `${theme.lifeStyleResultWarning}`,
    VeryLow: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'exercise',
    color: exerciseRiskMapping[exerciseRisk],
    CardIcon: <ExerciseIcon color={exerciseRiskMapping[exerciseRisk]} />,
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.exerciseRiskCard.title',
      'Exercise',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.exerciseRiskCard.subTitle.${exerciseRisk}`,
      exerciseRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.exerciseRiskCard.description.${exerciseRisk}`,
      exerciseRisk,
    ),
  };
};

export default createExerciseRiskInfo;
