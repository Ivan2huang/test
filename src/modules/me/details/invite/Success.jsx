import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'next/router';
import { injectIntl } from 'react-intl';
import { compose } from 'redux';
import { Box, Button, Card, CardContent } from '@material-ui/core';
import Images from '../../../../constants/images';
import { formatMessage, navigateTo } from '../../../../helpers/helpers';
import paths from '../../../../helpers/paths';
import Typography from '../../../../uiComponents/Typography';
import Grid from '../../../../uiComponents/Grid';
import GridItem from '../../../../uiComponents/GridItem';
import NavigationBackButton from '../../../../uiComponents/NavigationBackButton';
import { DETAILS_ID } from '../../constant';

const Success = ({ intl }) => {
  const navigateToDetailsPage = () => navigateTo(paths.common[DETAILS_ID]);

  return (
    <Box
      mt={{
        xs: 3,
        sm: 3,
      }}
    >
      <NavigationBackButton
        testId="back-btn-navigate-details-page"
        onClick={navigateToDetailsPage}
      >
        {formatMessage(
          intl,
          'me.tabs.myDetails.invite.dependent.button.backToMyDetails',
          '< My details',
        )}
      </NavigationBackButton>

      <Card>
        <CardContent>
          <Box mt={{ xs: 0, md: 10 }} mb={10}>
            <Grid>
              <GridItem offset={{ md: 1 }} columns={{ md: 3 }}>
                <img src={Images.INVITE_DEPENDENT_SUCCESS} alt="background" />
              </GridItem>
              <GridItem
                offset={{ sm: 1, md: 1 }}
                columns={{ xs: 12, sm: 10, md: 5 }}
              >
                <Box mb={6}>
                  <Typography type="style-1">
                    {formatMessage(
                      intl,
                      'me.tabs.myDetails.invite.dependent.header.success.title',
                      'Invite sent',
                    )}
                  </Typography>
                </Box>
                <Box mb={4}>
                  <Typography type="style-6">
                    {formatMessage(
                      intl,
                      'me.tabs.myDetails.invite.dependent.header.success.message',
                      'Ask your dependent to check their email for instructions to create their account',
                    )}
                  </Typography>
                </Box>
                <Box mb={8}>
                  <Button
                    type="submit"
                    variant="contained"
                    color="primary"
                    data-testid="btn-navigate-details-page"
                    onClick={navigateToDetailsPage}
                  >
                    {formatMessage(
                      intl,
                      'me.tabs.myDetails.invite.dependent.button.success',
                      'Back to my details',
                    )}
                  </Button>
                </Box>
              </GridItem>
            </Grid>
          </Box>
        </CardContent>
      </Card>
    </Box>
  );
};

Success.propTypes = {
  intl: PropTypes.shape({}).isRequired,
  router: PropTypes.shape({
    query: PropTypes.shape({
      dependentName: PropTypes.string,
      dependentEmail: PropTypes.string,
    }),
  }),
};

Success.defaultProps = {
  router: {},
};

export default compose(
  injectIntl,
  withRouter,
)(Success);
