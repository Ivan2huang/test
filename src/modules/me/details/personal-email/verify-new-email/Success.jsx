import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import LayoutContent from './LayoutContent';
import { formatMessage } from '../../../../../helpers/helpers';
import Images from '../../../../../constants/images';

const Success = ({ intl }) => {
  const title = formatMessage(
    intl,
    'me.details.verifyEmailSuccess.label.header',
    'Successfully Verified',
  );

  const description = formatMessage(
    intl,
    'me.details.verifyEmailSuccess.label.content',
    'Launch the app to login with your new email address.',
  );
  return (
    <LayoutContent
      title={title}
      description={description}
      backgroundImage={Images.VERIFY_PERSONAL_EMAIL_SUCCESS}
    />
  );
};

Success.propTypes = {
  intl: PropTypes.shape({}).isRequired,
};

export default injectIntl(Success);
