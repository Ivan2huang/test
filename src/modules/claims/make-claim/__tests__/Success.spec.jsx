/* eslint-disable react/prop-types */

import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import Success from '../Success';
import withTheme from '../../../../themes/withThemeProvider';
import { navigateTo } from '../../../../helpers/helpers';
import withIntl from '../../../../i18n/withIntlProvider';

jest.mock(
  '../../../../uiComponents/Typography',
  () => ({ children, ...rest }) => <div {...rest}>{children}</div>,
);

jest.mock(
  '../../../../uiComponents/ButtonGroup',
  () => ({ children, ...rest }) => <div {...rest}>{children}</div>,
);

jest.mock('../../../../helpers/helpers', () => ({
  navigateTo: jest.fn(),
  formatMessage: jest.fn((intl, id, defaultMessage) => defaultMessage),
}));

jest.mock('../../../../helpers/mui', () => ({
  useBreakpoint: jest.fn(() => 'xs'),
}));

describe('Success component', () => {
  let result;

  beforeEach(() => {
    const Component = withIntl(withTheme(Success));
    result = render(<Component />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the Snapshot', () => {
    expect(result.container).toMatchSnapshot();
  });

  it('should navigate to Make new claim when make another claim is clicked', () => {
    const makeNewClaimButton = result.getByTestId('btn-make-another-claim');
    fireEvent.click(makeNewClaimButton);

    expect(navigateTo).toHaveBeenCalledTimes(1);
    expect(navigateTo).toHaveBeenCalledWith('/claims/make-claim');
  });

  it('should navigate to claims when back to claims button is clicked', () => {
    const backToClaims = result.getByTestId('btn-back-to-claims');
    fireEvent.click(backToClaims);

    expect(navigateTo).toHaveBeenCalledTimes(1);
    expect(navigateTo).toHaveBeenCalledWith('/claims');
  });
});
