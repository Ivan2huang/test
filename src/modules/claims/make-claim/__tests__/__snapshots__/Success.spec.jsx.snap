// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success component should match the Snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-3"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <div
          class="MuiBox-root MuiBox-root-33"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiBox-root MuiBox-root-123"
              >
                <div
                  type="style-1"
                >
                  Claim submitted
                </div>
              </div>
              <div
                class="MuiBox-root MuiBox-root-124"
              >
                <div
                  type="style-6"
                >
                  Thank you. Please keep your original documents for at least 6 months. Your claim will be settled in 14 days.
                </div>
              </div>
              <div
                class="MuiBox-root MuiBox-root-125"
              >
                <div
                  type="style-6"
                >
                  Your claim might take 2-3 days before it shows up in the claims list.
                </div>
              </div>
              <div>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                  data-testid="btn-back-to-claims"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Back to claims
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
                  data-testid="btn-make-another-claim"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Make another claim
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
